From 0e2e06cee6edc3e3b8dede8500b627be2e30f91f Mon Sep 17 00:00:00 2001
From: Alex Johnston <acjohnston@google.com>
Date: Thu, 3 Mar 2022 14:30:44 +0000
Subject: [PATCH] Change default USB configuration to a RestrictedPreference

Test: manual with TestDPC and Settings
      DefaultUsbConfigurationPreferenceControllerTest
Bug: 201519976
     205996517
Change-Id: I1def7b37184d6d81f29a5e6e4793b92012dacd9b
Merged-In: I1def7b37184d6d81f29a5e6e4793b92012dacd9b
(cherry picked from commit bc4fd99ddb494683420264e6b749a5ce80df90b9)
Merged-In: I1def7b37184d6d81f29a5e6e4793b92012dacd9b
---
 res/xml/development_settings.xml                              | 2 +-
 .../DefaultUsbConfigurationPreferenceController.java          | 4 ++--
 .../DefaultUsbConfigurationPreferenceControllerTest.java      | 4 ++--
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/res/xml/development_settings.xml b/res/xml/development_settings.xml
index 621351681c..e389747bf6 100644
--- a/res/xml/development_settings.xml
+++ b/res/xml/development_settings.xml
@@ -293,7 +293,7 @@
             android:title="@string/tethering_hardware_offload"
             android:summary="@string/tethering_hardware_offload_summary" />
 
-        <com.android.settingslib.RestrictedSwitchPreference
+        <com.android.settingslib.RestrictedPreference
             android:key="default_usb_configuration"
             android:fragment="com.android.settings.connecteddevice.usb.UsbDefaultFragment"
             android:title="@string/usb_default_label"/>
diff --git a/src/com/android/settings/development/DefaultUsbConfigurationPreferenceController.java b/src/com/android/settings/development/DefaultUsbConfigurationPreferenceController.java
index be7704fd73..7c3d3b120e 100644
--- a/src/com/android/settings/development/DefaultUsbConfigurationPreferenceController.java
+++ b/src/com/android/settings/development/DefaultUsbConfigurationPreferenceController.java
@@ -24,7 +24,7 @@ import android.os.UserHandle;
 import androidx.preference.Preference;
 import androidx.preference.PreferenceScreen;
 
-import com.android.settingslib.RestrictedSwitchPreference;
+import com.android.settingslib.RestrictedPreference;
 import com.android.settingslib.development.DeveloperOptionsPreferenceController;
 
 public class DefaultUsbConfigurationPreferenceController extends
@@ -32,7 +32,7 @@ public class DefaultUsbConfigurationPreferenceController extends
 
     private static final String PREFERENCE_KEY = "default_usb_configuration";
 
-    private RestrictedSwitchPreference mPreference;
+    private RestrictedPreference mPreference;
 
     public DefaultUsbConfigurationPreferenceController(Context context) {
         super(context);
diff --git a/tests/robotests/src/com/android/settings/development/DefaultUsbConfigurationPreferenceControllerTest.java b/tests/robotests/src/com/android/settings/development/DefaultUsbConfigurationPreferenceControllerTest.java
index c9b13e27a0..a386473020 100644
--- a/tests/robotests/src/com/android/settings/development/DefaultUsbConfigurationPreferenceControllerTest.java
+++ b/tests/robotests/src/com/android/settings/development/DefaultUsbConfigurationPreferenceControllerTest.java
@@ -32,7 +32,7 @@ import android.os.UserHandle;
 import androidx.preference.PreferenceScreen;
 
 import com.android.settingslib.RestrictedLockUtils;
-import com.android.settingslib.RestrictedSwitchPreference;
+import com.android.settingslib.RestrictedPreference;
 
 import org.junit.Before;
 import org.junit.Test;
@@ -48,7 +48,7 @@ public class DefaultUsbConfigurationPreferenceControllerTest {
     private static final ComponentName TEST_COMPONENT_NAME = new ComponentName("test", "test");
 
     @Mock
-    private RestrictedSwitchPreference mPreference;
+    private RestrictedPreference mPreference;
     @Mock
     private PreferenceScreen mPreferenceScreen;
     @Mock
-- 
2.35.1.894.gb6a874cedc-goog

